packages/core/src/external/carequality/dq/process-inbound-dq.ts

```diff
@@ -1,11 +1,44 @@
+import { DocumentReference } from "@medplum/fhirtypes";
 import { InboundDocumentQueryReq, InboundDocumentQueryResp } from "@metriport/ihe-gateway-sdk";
-import { constructDQErrorResponse, IHEGatewayError, XDSRegistryError } from "../error";
-import { findDocumentReferences } from "./find-document-reference";
+import axios from "axios";
+import {
+  createUploadFilePath,
+  createUploadMetadataFilePath,
+} from "../../../domain/document/upload";
+import { Config } from "../../../util/config";
+import { out } from "../../../util/log";
+import { capture } from "../../../util/notifications";
+import { sizeInBytes } from "../../../util/string";
+import { uuidv7 } from "../../../util/uuid-v7";
+import { createAndUploadMetadataFile } from "../../aws/lambda-logic/document-uploader";
+import { S3Utils } from "../../aws/s3";
+import {
+  IHEGatewayError,
+  XDSRegistryError,
+  XDSUnknownPatientId,
+  constructDQErrorResponse,
+} from "../error";
+import { decodePatientId, findDocumentReferences } from "./find-document-reference";
+
+const region = Config.getAWSRegion();
+const s3Utils = new S3Utils(region);
+const api = axios.create();
+const bucket = Config.getMedicalDocumentsBucketName();
 
 export async function processInboundDocumentQuery(
-  payload: InboundDocumentQueryReq
+  payload: InboundDocumentQueryReq,
+  apiUrl: string
 ): Promise<InboundDocumentQueryResp> {
   try {
+    const id_pair = decodePatientId(payload.externalGatewayPatient.id);
+
+    if (!id_pair) {
+      throw new XDSUnknownPatientId("Patient ID is not valid");
+    }
+    const { cxId, id: patientId } = id_pair;
+
+    const endpointUrl = `${apiUrl}/internal/docs/ccd`;
+    await createAndUploadCcdAndMetadata(cxId, patientId, endpointUrl);
```

Comment:
If I understand this, we are uploading empty documents for every single patient? Shouldn't we only upload if the patient has no documents?

Reply:
And then for the metadata, just return it directly as opposed to reading it from s3?

Reply:
Why don't we want to upload the metadata file?

Reply:
>Shouldn't we only upload if the patient has no documents? 
Yes, good point. I forgot about that.

Reply:
https://github.com/metriport/metriport-internal/issues/1674 we have had a lot of issues with generating the metadata and storing it and it would be good to not double down on that framework

> Why don't we want to upload the metadata file?